qiskit_machine_learning/primitives/kernels/pseudo_kernel.py

```diff
@@ -0,0 +1,136 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Pseudo Overlap Kernel"""
+
+from typing import Optional, Callable, Union, List
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.primitives import Sampler
+from qiskit.primitives.fidelity import BaseFidelity
+
+from qiskit_machine_learning.utils import make_2d
+from .quantum_kernel import QuantumKernel
+
+SamplerFactory = Callable[[List[QuantumCircuit]], Sampler]
+FidelityFactory = Callable[[List[QuantumCircuit], SamplerFactory], BaseFidelity]
```

Comment:
Is it possible to define these in one place, e.g. `BaseKernel`? They seem to be used in all child classes.